tests/reftests/resolve-variables.test

```diff
@@ -14,6 +14,7 @@ The following actions will be performed:
 
 <><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
 -> installed foo.1
+[WARNING] Undefined filter variable build in dependencies of foo.1
```

Comment:
Is that expected?

Reply:
yes, because we recompute and re-resolve at new switch state recording